protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/detector/XmpDetector.java

```diff
@@ -0,0 +1,254 @@
+* THE KRUPCZAK ORGANIZATION, LLC
+* http://www.krupczak.org/
+*/
+
+/**
+ * OpenNMS XMP Detector allows for discovery of service/protocols via 
+ * provisiond.  Our detector is pretty simple right now.  All it does is
+ * attempt to establish an XmpSession with a system and if it succeeds,
+ * it queries a few core MIB variables and returns success.
+ * Future enhancements could include more capability information like
+ * supported MIBs, modules/plugins, and various dependencies to
+ * determine if something is a server for some service (noted by an
+ * inbound dependency relationship).
+ */
+
+package org.opennms.protocols.xmp.detector;
+
+import java.net.InetAddress;
+import java.util.Date;
+import org.opennms.netmgt.provision.support.BasicDetector;
+import org.opennms.netmgt.provision.support.AbstractDetector;
+import org.opennms.netmgt.provision.ServiceDetector;
+import org.opennms.netmgt.provision.SyncServiceDetector;
+import org.krupczak.xmp.*;
+import org.opennms.netmgt.config.xmpConfig.XmpConfig;    
+import org.opennms.netmgt.protocols.xmp.config.XmpConfigFactory;
+import org.opennms.netmgt.protocols.xmp.collector.XmpCollector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component
+/**
+ * XmpDetector class
+ * @author rdk <rdk@krupczak.org>
+ * @version $Id: $
+ *
+ */
+@Scope("prototype")
+public class XmpDetector implements SyncServiceDetector
+{
+    // class variables
+
+    private static final String DEFAULT_SERVICE_NAME = "XMP";
+    private static final Logger LOG = LoggerFactory.getLogger(XmpDetector.class);
+    private static int XMP_DEFAULT_TIMEOUT = 3000;
+    private static int XMP_DEFAULT_RETRIES = 1;
+    
+    // instance variables
+    SocketOpts sockopts;
+    XmpConfig protoConfig;
+    int xmpPort;
+    int xmpTimeout;
+    int xmpRetries;
+    String xmpAuthenUser;
+    String xmpServiceName; 
+    String m_ipMatch;
+    Date createTimeDate;
+
+    /* 
+     * @param serviceName a {@link java.lang.String} object
+     * @param port an int specifying tcp port number
+     */
+    
+    public XmpDetector(String serviceName, int port)
+    {
+	//super(serviceName,port,XMP_DEFAULT_TIMEOUT,XMP_DEFAULT_RETRIES);
+	
+        // set default config
+        xmpPort = port;
+        xmpAuthenUser = new String("xmpUser"); 
+        xmpTimeout = XMP_DEFAULT_TIMEOUT; /* millseconds */
+        xmpRetries = XMP_DEFAULT_RETRIES;
+
+        // get socket opts
+        sockopts = new SocketOpts();
+
+        xmpServiceName = new String(serviceName);
+
+	// very important to set to null, not 0-len string
+	// as provisiond's ip range matching functionality
+	// will not correctly match if 0-len instead of null
+	
+	//m_ipMatch = new String("");
+	m_ipMatch = null;  
+
+	createTimeDate = new Date();
+
+	if (LOG == null) {
+	   System.out.println("XmpDetector created, but null LOG");
```

Comment:
Hi!

I'm OK with however you guys want to mangle the submission.  When I
was originally writing the detector, I seem to recall seeing some
other detector issue a simlilar statement.

Bobby

>    In
>    [1]protocols/xmp/src/main/java/org/opennms/netmgt/protocols/xmp/detecto
>    r/XmpDetector.java:
> > +        // get socket opts
> > +        sockopts = new SocketOpts();
> > +
> > +        xmpServiceName = new String(serviceName);
> > +
> > +     // very important to set to null, not 0-len string
> > +     // as provisiond's ip range matching functionality
> > +     // will not correctly match if 0-len instead of null
> > +
> > +     //m_ipMatch = new String("");
> > +     m_ipMatch = null;
> > +
> > +     createTimeDate = new Date();
> > +
> > +     if (LOG == null) {
> > +        System.out.println("XmpDetector created, but null LOG");
> 
>    If the logger returned by the LoggerFactory is ever be null, it should
>    be considered a bug in slf4j, which doesn't justify the
>    system.out.println. I think this is something what Jeff means, right?
> 
>    —
>    Reply to this email directly or [2]view it on GitHub.
> 
> References
> 
>    1. https://github.com/OpenNMS/opennms/pull/215#discussion_r24007850
>    2. https://github.com/OpenNMS/opennms/pull/215/files#r24007850